src/server/error/error.go

```diff
@@ -51,3 +47,33 @@ func getHTTPStatusCode(errCode string) int {
 	}
 	return statusCode
 }
+
+var _ middleware.Responder = &ErrResponder{}
+
+// ErrResponder error responder
+type ErrResponder struct {
+	err error
+}
+
+// WriteResponse ...
+func (r *ErrResponder) WriteResponse(rw http.ResponseWriter, producer runtime.Producer) {
+	code := ierror.ErrCode(r.err)
+	rw.WriteHeader(getHTTPStatusCode(code))
+
+	var e *ierror.Error
+	if !errors.As(r.err, &e) {
+		e = &ierror.Error{
```

Comment:
The response should be an error array, is this covered by other logic?

Reply:
No, management APIs (v2.0) not needed keep the same as the oci error response.

Reply:
I think @wy65701436 has already implemented to return the error array for Harbor APIs in PR https://github.com/goharbor/harbor/pull/10324?

I'm OK for both these two solutions, but we must be consistent.